### ISSUE TITLE
Deleting a node server also uninstalls it.

### ISSUE THREAD

**Issue by bpaauwe** [UniversalDevicesInc/pg3#24]
I'm assuming this is by design but it's going to make node server development a lot more difficult.  A common practice when developing a node server is to delete it and re-add as there are certain actions that only happen when first starting a node server (I.E. writing the initial profile files to the ISY) that are going to be very difficult to do if the Polyglot delete now removes the entire directory.

I just lost all the modifications I had done to make my node server start on PG3.  The steps to reproduce this are:

1. install node server from store.  
2. node server fails to start at all and no profile files are written to ISY
3. modify node server so that it starts and runs, but PG3 doesn't think it needs to send the profile files
4. delete node server
5. all changes lost

I realize that removing the install/add delete/uninstall process and just doing install/delete is probably a better user experience but developers need a way around this.  We need a way to install development versions and a way to add/delete that from the Polyglot database.  Also, this change will break any node server designs that saved configuration files with the expectation that a user could delete and re-add without having to re-configure.

**Comment by mkohanim** [comment]
Bob,

I think we should have two differnet flows: one for development and one for end users. For end users, the flow you are describing is the correct one because we have to remove everything. For developers, though, we need to have a separate path. Very much like the APK for Android.

With kind regards,

******************************
  Michel Kohanim
  CEO

  (p) 818.631.0333
  http://www.universal-devices.com<http://www.universal-devices.com/>
******************************

**Comment by bpaauwe** [comment]
If everything is going to be removed, will there be a way to save node server configuration information?  

I believe there are node servers today that save configuration files to the node server directory with the expectation that you can delete/re-add and not have to re-configure. Is that no longer going to be possible?

**Comment by mkohanim** [comment]
Hi Bob, not so sure. Do we want to keep sensitive information (password/etc.) around?

With kind regards,

******************************
  Michel Kohanim
  CEO

  (p) 818.631.0333
  http://www.universal-devices.com<http://www.universal-devices.com/>
******************************

**Comment by jimboca** [comment]
Yes, harmony nodeserver has a bunch of files containing the harmony configuration which are saved to the server.json defined:
`    "persist_folder": "config",
`
These are large xml files pulled from the Harmony hubs so I didn't think storing them in the DB was a good idea.  So if these are deleted it could cause issues with nodes referenced in scenes or programs when reinstalled.

**Comment by bobpaauwe** [comment]
Yeah, that was the use case I was worried about. Even saving to the DB doesn't help here because those entries would be deleted when the node server was deleted.

With PG2, delete was more of a reset-database.  Basically removing the it from the database and ISY and then it could be re-added without having to re-install. 

From an end-user perspective, combining delete and uninstall is probably a good idea.  We'll just need to work out if there needs to be some way to reset for cases where things get out of wack.  Maybe a maintenance type option on Polyglot that allows the user to remove database data and deletes the node server from the ISY.  Then when the node server is started, it's more like a clean install where it has to re-add to the ISY and set any custom data.
